common/nglinks.py

```diff
@@ -0,0 +1,116 @@
+"""Neuroglancer link functions."""
+import json
+import urllib
+from typing import Optional, Union
+from collections import OrderedDict
+from dataclasses import dataclass
+from enum import Enum
+
+
+class LayerType(Enum):
+    Image = "image"
+    Segmentation = "segmentation"
+
+
+@dataclass
+class Layer:
+    cloudpath: str
+    type: LayerType
+
+
+class ImageLayer(Layer):
+
+    def __init__(self, cloudpath: str):
+        self.cloudpath = cloudpath
+        self.type = LayerType.Image
+
+
+class SegLayer(Layer):
+
+    def __init__(self, cloudpath: str):
+        self.cloudpath = cloudpath
+        self.type = LayerType.Segmentation
+
+
+def generate_ng_payload(
+    layers: dict[str, Layer], center: Optional[tuple[int, int, int]] = None
+) -> str:
+    """Makes neuroglancer link payloads from mappings between layer names and paths.
+
+    Makes the viewer resolution equal to the "img" layer if it exists in layers.
+    Otherwise, it takes the first it can find.
+    """
+    assert len(layers) > 0, "empty layer paths"
+
+    ngl_layers = OrderedDict()
+    layers = layers.copy()  # for in-place modifications later
+
+    # default to first resolution we can find - update if we find an image
+    ng_resolution = dataset_resolution(list(layers.values())[0])
+
+    # Trying to find an image layer
+    base_img = layers.pop("img", None)
```

Comment:
Not sure if we should try harder here, considering layers like "image", "img_xxx" etc

Reply:
I added some regular expression matching for this. What do you think?

Reply:
I wonder if we should abuse the ordered implementation of dict and just take the first one as the base layer. Actually since the order of layers does matter in neuroglancer,  maybe it is better to use a list to represent the layers, not dict.

Reply:
In that idea, how would you determine the layer names? Include them as class attributes?

Reply:
Yes, that matches with neuroglancer as well. I do not think the neuroglancer link will have a lot of layers, a list will not cost too much overhead.

Reply:
Thanks

Reply:
This should be done now.